File: src/LeetCode281.cpp
class ZigzagIterator {

private:
    vector<int> v1;
    vector<int> v2;
    int i1;
    int i2;

public:
    ZigzagIterator(vector<int>& v1, vector<int>& v2) {
        this->v1 = v1;
        this->v2 = v2;
        i1 = 0;
        i2 = 0;
    }

    int next() {
        return (i1 < v1.size() && i1 <= i2 || i2 == v2.size()) ?
                v1[i1++] : v2[i2++];
    }

    bool hasNext() {
        return i1 < v1.size() || i2 < v2.size();
    }
};

File: src/LeetCode281.java
public class ZigzagIterator {

    private int i1;
    private int i2;
    private List<Integer> v1;
    private List<Integer> v2;

    public ZigzagIterator(List<Integer> v1, List<Integer> v2) {
        i1 = 0;
        i2 = 0;
        this.v1 = new ArrayList<>(v1);
        this.v2 = new ArrayList<>(v2);
    }

    public int next() {
        if (i1 < v1.size() && i1 <= i2 || i2 == v2.size()) {
            return v1.get(i1++);
        } else {
            return v2.get(i2++);
        }
    }

    public boolean hasNext() {
        return i1 < v1.size() || i2 < v2.size();
    }
}
